threads/src/main/java/org/jboss/as/threads/BoundedQueueThreadPoolResourceDefinition.java

```diff
@@ -48,46 +54,76 @@
         if (blocking) {
-            return create(type, ThreadsServices.STANDARD_THREAD_FACTORY_RESOLVER, null, ThreadsServices.EXECUTOR, registerRuntimeOnly);
+            return create(type, ThreadsServices.getThreadFactoryResolver(type), null, ThreadsServices.EXECUTOR, registerRuntimeOnly);
         } else {
-            return create(type, ThreadsServices.STANDARD_THREAD_FACTORY_RESOLVER, ThreadsServices.STANDARD_HANDOFF_EXECUTOR_RESOLVER,
+            return create(type, ThreadsServices.getThreadFactoryResolver(type), ThreadsServices.getHandoffExecutorResolver(type),
                     ThreadsServices.EXECUTOR, registerRuntimeOnly);
         }
     }
 
     public static BoundedQueueThreadPoolResourceDefinition create(String type, ThreadFactoryResolver threadFactoryResolver,
                                                                   HandoffExecutorResolver handoffExecutorResolver,
                                                                   ServiceName poolNameBase, boolean registerRuntimeOnly) {
+        return create(PathElement.pathElement(type), threadFactoryResolver, handoffExecutorResolver,
+                ThreadsServices.createCapability(type, ManagedQueueExecutorService.class), poolNameBase,
+                registerRuntimeOnly);
+    }
+
+    public static BoundedQueueThreadPoolResourceDefinition create(PathElement path, ThreadFactoryResolver threadFactoryResolver,
+                                                                  HandoffExecutorResolver handoffExecutorResolver,
+                                                                  RuntimeCapability<Void> capability,
+                                                                  ServiceName poolNameBase, boolean registerRuntimeOnly) {
         final boolean blocking = handoffExecutorResolver == null;
-        final String resolverPrefix = blocking ? CommonAttributes.BLOCKING_BOUNDED_QUEUE_THREAD_POOL : CommonAttributes.BOUNDED_QUEUE_THREAD_POOL;
-        final BoundedQueueThreadPoolAdd addHandler = new BoundedQueueThreadPoolAdd(blocking, threadFactoryResolver, handoffExecutorResolver, poolNameBase);
+        final String resolverPrefix = blocking ? BLOCKING_BOUNDED_QUEUE_THREAD_POOL : BOUNDED_QUEUE_THREAD_POOL;
+        final BoundedQueueThreadPoolAdd addHandler = new BoundedQueueThreadPoolAdd(blocking, threadFactoryResolver,
+                handoffExecutorResolver, poolNameBase, capability);
         final OperationStepHandler removeHandler = new BoundedQueueThreadPoolRemove(addHandler);
-        return new BoundedQueueThreadPoolResourceDefinition(blocking, registerRuntimeOnly, type, poolNameBase, resolverPrefix, addHandler, removeHandler);
+        return new BoundedQueueThreadPoolResourceDefinition(blocking, registerRuntimeOnly, path, capability,
+                poolNameBase, resolverPrefix, addHandler, removeHandler);
     }
-
     /**
+     * @param blocking
```

Comment:
I'm nitpicking but if there is no description associated to the javadoc parameters, I see no point to add them